src/codecs/csl/index.ts

```diff
@@ -0,0 +1,228 @@
+  const csl = csls[0]
+
+  return decodeCsl(csl)
+}
+
+/**
+ * Encode a `Node` to CSL-JSON or other bibliographic format
+ *
+ * See https://citation.js.org/api/tutorial-output_options.html
+ * for formats and other options which could be used.
+ */
+export const encode: Encode = async (
+  node: stencila.Node,
+  options: EncodeOptions = {}
+): Promise<vfile.VFile> => {
+  const { format = 'json' } = options
+
+  let content = ''
+  if (isCreativeWork(node)) {
+    const csl = encodeCsl(node)
+    const cite = new Cite([csl])
+    if (format === 'json') {
+      const { _graph, ...rest } = cite.data[0]
+      content = JSON.stringify(rest, null, 2)
+    } else {
+      content = cite.format(format)
+    }
+  } else {
+    errorNodeType('csl', 'encode', 'CreativeWork', node)
+  }
+
+  return vfile.load(content)
+}
+
+/**
+ * Decode `Csl.Data` to a `CreativeWork` or derived type.
+ */
+export async function decodeCsl(
+  csl: Csl.Data
+): Promise<stencila.CreativeWork | stencila.Article> {
+  const {
+    type,
+    id,
+
+    author = [],
+    title = '',
+
+    issued,
+
+    'container-title': containerTitle,
+    volume,
+    issue,
+    page,
+
+    // @ts-ignore // Citation.js uses this hidden property
+    _graph,
+
+    ...lost
+  } = csl
+
+  if (type === 'article-journal') {
+    warnLossIfAny('csl', 'decode', csl, lost)
+
+    const authors = await Promise.all(author.map(decodeAuthor))
+
+    const datePublished = issued !== undefined ? decodeDate(issued) : undefined
+
+    let isPartOf
+    if (containerTitle !== undefined) {
+      isPartOf = stencila.periodical({
+        title: containerTitle
+      })
+      if (volume !== undefined) {
+        isPartOf = stencila.publicationVolume({
+          volumeNumber: volume,
+          isPartOf
+        })
+      }
+      if (issue !== undefined) {
+        isPartOf = stencila.publicationIssue({
+          issueNumber: issue,
+          isPartOf
+        })
+      }
+    }
+
+    return stencila.article(authors, title, {
+      id,
+      datePublished,
+      isPartOf
+    })
+  } else {
+    warnLoss('csl', 'decode', `Unhandled citation type ${csl.type}`)
+    return stencila.creativeWork()
+  }
+}
+
+/**
+ * Encode a `CreativeWork` as `Csl.Data`
+ */
+const encodeCsl = (cw: stencila.CreativeWork): Csl.Data => {
+  const { title = 'Untitled', authors = [], ...lost } = cw
+
+  warnLossIfAny('csl', 'encode', cw, lost)
+  return {
+    type: 'article-journal',
+    id: 'id', // TODO id is required
+    title: title,
+    author: authors.map(encodeAuthor)
+  }
+}
+
+/**
+ * Decode a `Csl.Person` as a `Person`
+ *
+ * If `family` is not defined but `literal` is then the
+ * `person` codec is used to decode the literal string.
+ *
+ * CSL-JSON's `non-dropping-particle` and `dropping-particle`
+ * are not currently supported in `Person`.
+ */
+const decodeAuthor = async (author: Csl.Person): Promise<stencila.Person> => {
```

Comment:
Brief tl;dr type doc comments for each function - with more details if necessary. Don't worry about `@param` etc since TypeDoc does that for use / its self documenting having types.